lib/plugins/acts_as_journalized/lib/redmine/acts/journalized/save_hooks.rb

```diff
@@ -79,6 +81,21 @@ def add_journal(user = User.current, notes = '')
       @journal_notes ||= notes
     end
 
+    def save_journal_with_retry(journal, tries: 2)
+      begin
+        journal.save!
+      rescue ActiveRecord::RecordInvalid => e
+        # TODO: rescue from ActiveRecord::RecordNotUnique as well
+        if e.message =~ /Version has already been taken/ && tries > 0
+          journal.increment(:version)
+          tries -= 1
+          retry
+        else
+          raise
```

Comment:
I'm not actually sure that it's desirable to re-raise here. Not sure of the original reasons for #1224, but validation errors in journals probably shouldn't propagate up and cause obsure-to-debug validation errors on the journable model. 
Given that there is no special UI in place for a failed journal - so no action an end-user can actually take -, we might want to consider logging there instead.